executor/containerdexecutor/executor.go

```diff
@@ -118,7 +118,13 @@ func (w *containerdExecutor) Run(ctx context.Context, id string, root executor.M
 	}()
 
 	meta := process.Meta
-	resolvConf, hostsFile, releasers, err := w.prepareExecutionEnv(ctx, root, mounts, meta, details)
+
+	provider, ok := w.networkProviders[meta.NetMode]
+	if !ok {
+		return nil, errors.Errorf("unknown network mode %s", meta.NetMode)
+	}
+
+	resolvConf, hostsFile, releasers, err := w.prepareExecutionEnv(ctx, root, mounts, meta, details, provider.LocalhostAllowed())
```

Comment:
Not sure why this is called `LocalhostAllowed`.

I think DNS should depend on user provided value, not the implementation. If the user requests --net host` then it is fine to allow host DNS. But if user requests default net and `default==host` because no CNI or bridge is requested then it should be the default DNS. This also lets us switch the default provider (to bridge) and makes sure users do not depend on different behavior depending on how buildkit is deployed.